SWENT-GRP5/SOLVIT
# Enhancing UI Consistency and Backend Refinements

### Overview

This PR focuses on improving the overall color consistency across the app, addressing small UI and backend issues, and enhancing the user experience. Key updates include changes to button and selection colors, UI edits for improved aesthetics and clarity, as well as backend refactoring and feature improvements.

### UI Changes

**Color Updates:**

- AI Screen:
  - Changed button colors for better visual appeal and consistency.
<img width="331" alt="image" src="https://github.com/user-attachments/assets/873b98cb-7023-4b08-8032-1a64ba472012" />

- Provider Info 
 - Adjusted the color of packages when selected for improved clarity.
<img width="331" alt="image" src="https://github.com/user-attachments/assets/1a40bc8e-1b20-4324-adef-875a5fa19c6f" />

 
- Home Screen:
  - Updated the UI to display the slogan "Solvate" instead of the provider's name and address.
<img width="331" alt="image" src="https://github.com/user-attachments/assets/8631c66c-e6be-495c-8cdd-a1922077c240" />

- Select Provider Screen:
  - Changed the color of the "Select the best provider near you" text to white for better contrast.
  
<img width="331" alt="image" src="https://github.com/user-attachments/assets/f46a0191-f6a1-41bc-8723-85401f82989d" />

- Provider & Seeker Info Screens:
  - Edited and aligned design elements across both screens for a consistent look and feel.
- Dialog Updates:
  - Enhanced the dialog used for assigning requests to make it more visually appealing.
  
<img width="331" alt="image" src="https://github.com/user-attachments/assets/f14f103d-6eb8-42af-81fc-05ef98421cfc" />

- Booking info : 
  - Enhanced color of chosen package
  
<img width="331" alt="image" src="https://github.com/user-attachments/assets/cdd129e1-11ab-49a6-a55d-0d2b12764f8a" />


**Backend Changes**

- Field Refactoring:
  - Changed the address field from a String to a Location object to improve data representation and consistency.
- Filter Functionality Improvements:

- Fixed issues with provider filtering.
- Refactored the filtering logic to be managed within the ViewModel:
  - Users can now filter providers by location and other fields (e.g., ratings, languages) simultaneously.
  - Added tests to verify that the filtering functionality works seamlessly. (I encourage you to test this thoroughly.)
- Additional Notes

  - All changes aim to enhance the app's usability, maintainability, and aesthetic appeal.
  - Backend changes are backward-compatible with existing functionality but introduce better practices and improved user experiences.
### Testing Instructions

- Verify the updated colors and UI consistency across all mentioned screens.
- Test provider filtering:

  - Apply multiple filters (e.g., by location and rating) and confirm expected results.
  - Confirm the address field refactoring doesn’t break existing functionality or data processing.
